### PR TITLE
Workshop2024 merge

### LINKED ISSUE #2339: [TASK] Update workshop materials
--------
Issue Description
--------

**Is your feature request related to a problem? Please describe.**
Update raven workshop materials to align it with current devel branch and 3.1 release.

**Describe the solution you'd like**

Update slides and hands-on tests. 

**Describe alternatives you've considered**

----------------
For Change Control Board: Issue Review
----------------
This review should occur before any development is performed as a response to this issue.
- [x] 1. Is it tagged with a type: defect or task?
- [x] 2. Is it tagged with a priority: critical, normal or minor?
- [x] 3. If it will impact requirements or requirements tests, is it tagged with requirements?
- [x] 4. If it is a defect, can it cause wrong results for users? If so an email needs to be sent to the users.
- [x] 5. Is a rationale provided? (Such as explaining why the improvement is needed or why current code is wrong.)

-------
For Change Control Board: Issue Closure
-------
This review should occur when the issue is imminently going to be closed.
- [x] 1. If the issue is a defect, is the defect fixed?
- [x] 2. If the issue is a defect, is the defect tested for in the regression test system?  (If not explain why not.)
- [x] 3. If the issue can impact users, has an email to the users group been written (the email should specify if the defect impacts stable or master)?
- [x] 4. If the issue is a defect, does it impact the latest release branch? If yes, is there any issue tagged with release (create if needed)?
- [x] 5. If the issue is being closed without a pull request, has an explanation of why it is being closed been provided?

### PR DESCRIPTION
**Pull request by PaulTalbot-INL**
--------
Pull Request Description
--------
##### What issue does this change request address? (Use "#" before the issue to link it, i.e., #42.)
close #2339 

##### What are the significant changes in functionality due to this change request?

----------------
For Change Control Board: Change Request Review
----------------
The following review must be completed by an authorized member of the Change Control Board.
- [x] 1. Review all computer code.
- [x] 2. If any changes occur to the input syntax, there must be an accompanying change to the user manual and xsd schema. If the input syntax change deprecates existing input files, a conversion script needs to be added (see Conversion Scripts).
- [x] 3. Make sure the Python code and commenting standards are respected (camelBack, etc.) - See on the [wiki](https://github.com/idaholab/raven/wiki/RAVEN-Code-Standards#python) for details.
- [x] 4. Automated Tests should pass, including run_tests, pylint, manual building and xsd tests. If there are changes to Simulation.py or JobHandler.py the qsub tests must pass.
- [x] 5. If significant functionality is added, there must  be tests added to check this. Tests should cover all possible options.  Multiple short tests are preferred over one large test. If new development on the internal JobHandler parallel system is performed, a cluster test must be added setting, in <RunInfo> XML block, the node ```<internalParallel>``` to True.
- [x] 6. If the change modifies or adds a requirement or a requirement based test case, the Change Control Board's Chair or designee also needs to approve the change.  The requirements and the requirements test shall be in sync.
- [x] 7. The merge request must reference an issue.  If the issue is closed, the issue close checklist shall be done.
- [x] 8. If an analytic test is changed/added is the the analytic documentation updated/added?
- [x] 9. If any test used as a basis for documentation examples (currently found in `raven/tests/framework/user_guide` and `raven/docs/workshop`) have been changed, the associated documentation must be reviewed and assured the text matches the example.

### CODE REVIEW

**Comment by wangcj05** [comment]
@joshua-cogliati-inl It seems there is some problem with IDW rom: 
image

Do you have time to take a look at it? You can check test at:  

image

**Comment by joshua-cogliati-inl** [comment]
> @joshua-cogliati-inl It seems there is some problem with IDW rom: 

Hm, I wonder if commenting out line 234 in `crow/include/utilities/ND_Interpolation_Functions.h` would fix it.
(That is `virtual ~InverseDistanceWeighting() {};` )

@j-bryan (Any ideas?)

**Comment by j-bryan** [comment]
@joshua-cogliati-inl I tried commenting out that line, rebuilding RAVEN, and running the test, and I still got that warning.

**Review by joshua-cogliati-inl — CHANGES_REQUESTED** [review]
This pull request adds workshop materials, but it also makes changes in ravenframework (especially in Samplers), however, this is not explained in the Pull Request Description or in the linked Issue/Task.

The workshop additions and the workshop example additions look fine (tho' some of the powerpoint files are on the big side).

File: doc/workshop/SparseSensing/SparseSensing.pptx
[the remark this answers is not in the document]
Comment (Jimmy-INL) [reply doc/workshop/SparseSensing/SparseSensing.pptx:1]: I have removed few slides, but it is still 33 MB. I cannot further reduce it @joshua-cogliati-inl

[the remark this answers is not in the document]
Comment (alfoa) [reply doc/workshop/SparseSensing/SparseSensing.pptx:1]: To reduce the size, you can make power poin compress the pictures (for display usage) (maybe it does not impact the quality too much):
Screenshot 2024-07-10 at 5 12 19 PM

[the remark this answers is not in the document]
Comment (joshua-cogliati-inl) [reply doc/workshop/SparseSensing/SparseSensing.pptx:1]: For what it is worth, using compress with HD shrinks this by about a third, and "print" shrinks it to about half.

**Review by joshua-cogliati-inl — COMMENTED** [review]
The merge fixed the extra changes, so I no longer think this needs changes.

**Review by joshua-cogliati-inl — COMMENTED** [review]
I think it would be worth trying to compress some of the powerpoint presentations.

File: doc/workshop/advancedSampling/scripts/conv_data.xlsx
[the remark this answers is not in the document]
Comment (wangcj05) [reply doc/workshop/advancedSampling/scripts/conv_data.xlsx:1]: This is used in the slides for comparison.

**Comment by wangcj05** [comment]
@joshua-cogliati-inl FYI, I have reduced the sizes of slides based on your suggestion using `print` option.

**Comment by wangcj05** [comment]
This PR only touches workshop materials, and there is no source code updates in this PR. Since HPC is under maintenance, the Test qsubs sawtooth can be skipped in this PR. Once other tests are green, this PR can be merged.

**Pull request merged**